mmv1/third_party/terraform/services/storage/resource_storage_folder_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+package storage_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccStorageFolder_storageFolderBasic(t *testing.T) {
+	t.Parallel()
+
+	bucketName := acctest.TestBucketName(t)
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccStorageFolder_storageBucket(bucketName, true, true) + testAccStorageFolder_storageFolder(true),
+			},
+			{
+				ResourceName:            "google_storage_folder.folder",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"bucket", "recursive", "force_destroy"},
+			},
+		},
+	})
+}
+
+func TestAccStorageFolder_hnsDisabled(t *testing.T) {
+	t.Parallel()
+
+	bucketName := acctest.TestBucketName(t)
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccStorageFolder_storageBucket(bucketName, false, true) + testAccStorageFolder_storageFolder(true),
+				ExpectError: regexp.MustCompile("Error creating Folder: googleapi: Error 409: The bucket does not support hierarchical namespace., conflict"),
+			},
+		},
+	})
+}
+
+func TestAccStorageFolder_FolderForceDestroy(t *testing.T) {
```

Comment:
Just a intuition, I believe this test won't check the actual logic of force destroy deeply? The reason is we are creating all nested resources using terraform config only which will be part of terraform dependency graph. The deletion will happen by individual resource's DELETE method so there won't any object or sub-folder which will be deleted by force destroy. 
It would be good if we can some how add object in folder outside of terraform and then try to test force destroy functionality.

Reply:
objects are deleted only when force_destroy is set to true otherwise objects will not be deleted and will not  continue the delete operation